vcd/resource_vcd_org_user.go

```diff
@@ -233,10 +234,24 @@ func resourceToUserData(d *schema.ResourceData, meta interface{}) (*govcd.OrgUse
 func resourceToOrgUser(d *schema.ResourceData, meta interface{}) (*govcd.OrgUser, *govcd.AdminOrg, error) {
 
 	vcdClient := meta.(*VCDClient)
-	adminOrg, err := vcdClient.GetAdminOrgFromResource(d)
+	sessionInfo, err := vcdClient.Client.GetSessionInfo()
 	if err != nil {
-		return nil, nil, fmt.Errorf("[resourceToOrgUser] error retrieving org: %s", err)
+		return nil, nil, fmt.Errorf("error retrieving session info: %s", err)
 	}
+	result, err := runWithRetry("[resourceToOrgUser]",
```

Comment:
This PR contains almost ten new runWithRetry calls. All of them would benefit from a comment why are we doing this and what failure exactly we are targeting.

Speaking of which, what would happen if there's an error, but the error is not what we expect and it is an error which won't go away after retrying? Won't it cause the plan apply to stuck for the duration of the timeout? Right now it looks like we're "casting a wide net" here, which may catch true errors too. What about narrowing down the `runWithRetry` function to retry only on a specific condition/failure/error message? It doesn't have to be too specific, but could be a substring-match, which narrows down the scope.

Reply:
I tried to find a root cause, without avail.
If I has found such pattern, probably I would have used a different approach. The usage of `runWithRetry` happens because these failing calls only fail when we deal with shared catalog, and they manifest themselves as temporary access denied incidents, involving catalogs, users, and orgs. In each case, the error wording is different.
As it is now, the scope is the narrowest we could use.

Reply:
We had discussed this more, and there's just no clear way to narrow down the scope of these retries. Resolving this thread.